scripts/updatetranslations.py

```diff
@@ -0,0 +1,118 @@
+    proc = subprocess.Popen(["git", "reset", '--mixed'], stdout=subprocess.PIPE, shell=True)
+    proc.communicate()
+    if proc.returncode != 0:
+        logging.error('The git reset call returned an error status code ' + proc.returncode)
+        exit(1)
+    proc = subprocess.Popen(["git", "add", 'src/mumble/mumble_*.ts'], stdout=subprocess.PIPE, shell=True)
+    proc.communicate()
+    if proc.returncode != 0:
+        logging.error('The git add call returned an error status code ' + proc.returncode)
+        exit(1)
+    proc = subprocess.Popen(["git", "commit", '-m', 'TRANSLATION: Update translation files'], stdout=subprocess.PIPE, shell=True)
+    proc.communicate()
+    if proc.returncode != 0:
+        logging.error('The git commit call returned an error status code ' + proc.returncode)
+        exit(1)
+
+def Update(lupdatebin):
+    proc = subprocess.Popen([
+        lupdatebin
+        # Do not explain what is being done.
+        , "-silent"
+        , '-no-ui-lines'
+        # {sametext|similartext|number}
+        , '-disable-heuristic', 'similartext'
+        # {absolute|relative|none}
+        , '-locations', 'none'
+        , '-no-obsolete'
+        , '-no-recursive'
+        , '-extensions', 'ui,c,cpp,h,mm'
+        # sources
+        , './src', './src/mumble'
+        # target
+        , '-ts', tsfile
+        ], stdout=subprocess.PIPE, shell=True)
+    proc.communicate()
+    if proc.returncode != 0:
+        logging.error('lupdate failed with error code %d', proc.returncode)
+        exit(1)
+
+###
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--vcpkg-triplet', type=str, required=False, help='the vcpkg triplet to use the lupdate tool from, e.g. "x64-windows-static-md"')
+parser.add_argument('--debug', dest='debug', action='store_true')
+parser.set_defaults(debug=False)
+args = parser.parse_args()
+
+loglevel = logging.DEBUG if args.debug else logging.INFO
+logging.basicConfig(stream=sys.stdout, level=loglevel, format='%(levelname)s: %(message)s')
+
+lupdatebin = FindLupdate(args.vcpkg_triplet)
+if lupdatebin is None:
+    logging.error('Could not find `lupdate` executable')
+    exit(1)
+
+# cd into repository root directory
+os.chdir(os.path.dirname(os.path.abspath(__file__)) + "/..")
+logging.info('Working in directory %s', os.getcwd())
+
+logging.info('Ensuring ts files have no local changes…')
+if CheckForGitHasTsFileChanges():
+    logging.error('There are ts files with local changes. Cancelling execution.')
+    exit(1)
+
+tsfiles = glob.glob(os.path.join('src', 'mumble', 'mumble_*.ts'))
+logging.debug('Identified ts files: %s', tsfiles)
+
+logging.info('Updating ts files…')
+for tsfile in tsfiles:
+    logging.debug('Updating ts file ' + tsfile + '…')
+    Update(lupdatebin)
+
+if CheckForGitHasTsFileChanges():
+    logging.info('Committing changes…')
+    Commit()
+else:
+    logging.info('No changes to commit.')
+
+logging.info('The updating of the translation files successfully completed.')

```

Comment:
I am generally a fan of encapsulating the actual code into a `main()` function and the explicitly checking for `__name__ == "__main__"` and only then calling `main()`. In this case it probably doesn't matter that much but I think it is generally a good pattern to follow.

Reply:
What’s that ugly if for? So the file can be included or imported elsewhere without running code?

Reply:
Yes: https://stackoverflow.com/questions/419163/what-does-if-name-main-do